### ISSUE TITLE
Unable to build rom after creating new maps - map_group_count.h not found

### ISSUE THREAD

**Issue by c-a-i-o-v** [huderlem/porymap#635]
Hello!

I am deep into learning how to make changes using the pokeemerald-expansion decomp.

DAY 1-
After some changes to the early game, I thought it would be good to "make" rom and test if I was going in the right direction.
I had some problems trying to make the rom, but was able to move past them after reviewing everything edited and correcting some mistakes in the code. However, I don't know how to deal with the following error:

"fatal error: data/map_gourp_count.h: No such file or directory"

I tried to find the solution everywhere, but did not find anyone commenting about this.

The "map_group_count.h" file don't even exist in the master branch, so I am lost.

Things I changed during editing:
- Created new route east of Oldale Town (Using Porymap, following all instructions)
- Added 1 trainer to the route (had an error about this when compiling, but moved past it)
- Added some wild encounters to the route
- Added some berries to it.

DAY 2-
So, I was able to PARTIALLY figure it out yesterday.

It seems the map_group_count.h isn't being generated because all maps on data>maps had their headers and events files erased after I opened the project using porymap. But I don't know why that happens.
I download a clean version of the pokeemerald repository, did not create any new maps, just messed with some route 1 encounters, and then tried to make the rom. It worked.

Now, when I go to the github online pokeemerald-expansion repository, the data>maps folders don't have headers and events files.
But when I download it, these files come together.

New maps I made with porymap did not come with these files.
Not only that, but simple fact of using porymap to create a new map, delete these files (headers and events) from the original maps' folders.

It is very weird, and I don't know how to fully solve this problem. I am kinda locked of using porymap.

Do you have any idea about it?

**Comment by GriffinRichards** [comment]
Hello, thanks for your detailed report.
- That `map_group_count.h` file is a generated file unique to pokeemerald-expansion, Porymap isn't responsible for generating it and it doesn't interact with it all. If it's not being generated something else has presumably gone wrong in your project.
- Those `header.inc` and `events.inc` files are also generated, so those files being missing in the folders for new maps (or existing maps in a freshly cloned project) is normal and ok. They will be generated if/when your project builds successfully. Porymap likewise does not interact with these files at all, it's not responsible for generating them and it won't ever delete them.

It sounds to me like you have some problem in your project that's unrelated to Porymap (at least not directly). If you can either post a GitHub link to your up-to-date project or join the pret Discord server I or someone else can try to help you diagnose what that problem is. If it's a problem specific to pokeemerald-expansion you can also find help in their channels of the RHH Discord server

**Comment by c-a-i-o-v** [comment]
@GriffinRichards 

I can't join the discord server because the link has expired, or too many people already in the server.
I have tried to join there.

Maybe I am doing something wrong when creating new maps?
Because when I clone the project, and instantly try to build rom, I manage to do it properly. No problems there.
I have been testing with different editings to understand where things go wrong. 

On one of my tests, I used porymap to only edit route 101 encounters, and also create a map south of Petalburg (Inside petalburg's city group of maps).
On that test, I was able to build the rom. Everything was okay, the encounters had been edited and I could walk to the map south of petalburg (blank map, only had to remove some trees on Petalburg).

But all other tests did not build the rom, always showing the same error message (map_group_count not found).
One of them was a very simple editing, where I did not create any new map, just created some wild encounters in Oldale. 

All of this make me very confused kkkk
- I was able to edit encounters on created maps.
- I wasn't able to create encounters on maps that already exist.
- I was able to create a new map on a City group map.
- I wasn't able to create a new map on Route group maps.

**Comment by c-a-i-o-v** [comment]
I got an answer on Pokecommunity about this.
A now friend called Lunos investigated this same problem. I will paste his answer down below, but it seems it was not a problem with porymap.

From LUNOS, at pokecommunity
"When you download (as in git clone[I])[/I] the Pokeemerald-expansion there are no extra files in any of the maps' folders. What you see on GitHub is exactly what you get.
New files are generated as a part of the building-a-ROM process. Those files are built from scratch by the different tools the project uses.

That aside, I just cloned a fresh copy of the expansion, and to my surprise I got the exact same error you got about missing a data/map_group_count.h file.
This is most likely related to a problem in the project that has to be solved by its mantainers.
To solve it manually just make an empty copy of src/data/map_group_count.h, throw a make clean on your terminal and then make,
The project will build normally after that. I made a small change to Littleroot Town and a new map connected to it, and all that worked with no issues."


So he had the same information I gave here. He also mentioned about some files being created when building the rom.

<img width="1324" alt="Screenshot 2024-11-03 at 18 30 59" src="https://github.com/user-attachments/assets/212b63f5-109a-407e-9d1a-5cf1a987dd8b">
